dartagnan/src/main/java/com/dat3m/dartagnan/expression/type/TypeFactory.java

```diff
@@ -11,8 +11,6 @@
 
 public final class TypeFactory {
```

Comment:
With `TypeFactory` not being (treated as) a static class anymore, I think it might be a good idea to have each `Type` know its `TypeFactory`. It at least enables some type checks be made whenever an expression is given. For example in `Alloc`'s constructor you needed to get rid of the type precondition due to the lack of factory access.
Alternatively, it might just be better to put those preconditions that we cannot check inside the constructor anymore into the `EventFactory`.

Reply:
I have added the check to `EventFactory`.  We can add a `Type.getFactory()` later, should we ever need it.